### ISSUE TITLE
Liveness probes

### ISSUE THREAD

**Issue by nicolabovolato** [Julien-R44/bentocache#9]
Hi and thank you for this much needed library.

Have you considered, or are you planning to add, a way to drivers are able to communicate properly?
I'm thinking about exposing a function to `ping` redis or `select 1` a database.

It would be great to have such functionality available on the BentoCache instance, to be called when needed or exposed by an http route.

**Comment by Julien-R44** [comment]
hey, thanks !

i am not sure if that's the role of the library to do that

when you create your Bentocache instance, you must pass your `ioredis` instance, for example. so you can keep this instance around and use it for your route + health check. so yeah, nothing stops you from doing this without bentocache exposing this functionality. It just seems a bit out of scope to me

and also, there are drivers for which it wouldn't make sense to have this, like filesystem and in-memory.

let me know what you think

**Comment by nicolabovolato** [comment]
I agree with you that if a library like bentocache provides one with the ability to supply his own (redis or whatsoever) instances there should be no need to implement healthchecks. Still I think that given the aim of bentocache, a built in healthcheck would be nice!

Moreover, I don't see this capability with redis' bus driver, as in the constructor you can only provide ioredis options.
I could always wrap or extend the driver class if it wasn't that (and rightly so) underlying instances are private members.

As for memory or fs drivers, I still think that a healthcheck can [[1]](https://nodejs.org/api/os.html#osfreemem) [[2]](https://nodejs.org/api/fs.html#fspromisesstatfspath-options) be implemented, even if that may not make sense for common workloads. Also, testing won't be trivial.

With that being said, I still think it would be a nice feature but perhaps not as valuable as it is to me for other users.
Up to you! I would be happy to submit a pr should ever change your mind.

**Comment by Julien-R44** [comment]
Closing the issue since I don't have the bandwith to work on it right now. If the feature is still needed feel free to reopen it
